src/common/gpx.c

```diff
@@ -173,13 +259,78 @@ gboolean dt_gpx_get_location(struct dt_gpx_t *gpx, GDateTime *timestamp, dt_imag
+    const gint cmp_n = g_date_time_compare(timestamp, tp_next->time);
     if((cmp >= 0) && (item->next && cmp_n <= 0))
     {
-      geoloc->longitude = tp->longitude;
-      geoloc->latitude = tp->latitude;
-      geoloc->elevation = tp->elevation;
+      GTimeSpan seg_diff = g_date_time_difference(tp_next->time, tp->time);
+      GTimeSpan diff = g_date_time_difference(timestamp, tp->time);
+      if (seg_diff == 0)
+      {
+        geoloc->longitude = tp->longitude;
+        geoloc->latitude = tp->latitude;
+        geoloc->elevation = tp->elevation;
+      }
+      else
+      {
+        /* get the point by interpolation according to timestamp
+
+        We assume that the maximum difference in longitude is less or equal 180º:
+        since the bigger use case is that of an airplane, never an airplane flies more than 180º in longitude */
+
+        const double lat1 = tp->latitude;
+        const double lon1 = tp->longitude;
+        const double lat2 = tp_next->latitude;
+        const double lon2 = tp_next->longitude;
+
+        double lat, lon;
+
+        double f = (double)diff / (double)seg_diff; /* the fraction of the distance */
+
+        if (
+          fabs(lat2 - lat1) < DT_MINIMUM_ANGULAR_DELTA_FOR_GEODESIC &&
+          fabs(lon2 - lon1) < DT_MINIMUM_ANGULAR_DELTA_FOR_GEODESIC
+        )
+        {
+          /* short distance (< 10 km), no need for geodesic interpolation */
+          lon = lon1 + (lat2 - lon1) * f;
+          lat = lat1 + (lat2 - lat1) * f;
+        }
+        else
+        {
+          /* interpolation on the earth surface
+             formulas from http://www.movable-type.co.uk/scripts/latlong.html
+
+             the formulas are correct even if the two point are across the day line, e.g [(0, -179), (0,179)]
+             TO DO: in this case the line which is drawn is incorrect, but this should be a osm_gps issue
+          */
+
+          /* first, calculate the distance on the earth surface */
+          double d, delta;
+          dt_gpx_geodesic_distance(lat1, lon1,
+                                   lat2, lon2,
+                                   &d, &delta
+                                  );
+          /* d is the distance on the surface in metres,
+             delta is the angle defined by the two points*/
+
+          /* then, calculate the intermediate point */
+          dt_gpx_geodesic_intermediate_point(lat1, lon1,
+                                             lat2, lon2,
+                                             delta,
+                                             TRUE,
+                                             f,
+                                             &lat, &lon
+                                            );
+        }
+
+        geoloc->latitude = lat;
+        geoloc->longitude = lon;
+
+        /* make a simple linear interpolation on elevation */
+        geoloc->elevation = tp->elevation + (tp_next->elevation - tp->elevation) * f;
```

Comment:
Would make more sense for short distance, as suggested by @wpferguson .
Here I would set elevation = NAN to make clear it is not relevant.

Reply:
this calculation of interpolated elevation is made for short _and_ long distances

setting `elevation = NAN` makes sense only if one of the two is NAN.

Reply:
> this calculation of interpolated elevation is made for short and long distances

what is the meaning of average elevation on long distance ?

Reply:
> what is the meaning of average elevation on long distance ?

well, it's only a guess, but I cannot see a better one

Reply:
I really prefer a NAN than a wrong information which looks like a good one. This way I can fix manually if I have it.
In your map example (4 points), are the elevations meaningful in some way ?

Reply:
Looking at my use cases of #9950:
- in the first the two solutions could make sense, and anyway the user would have the possibility (if able to do it) to modify the elevation
- in the second case the interpolation surely makes sense, while NAN doesn't.

The first case is a quite rare one.

Reply:
> in the first the two solutions could make sense, and anyway the user would have the possibility (if able to do it) to modify the elevation

Some elevations could make sense, yes. In that sense it may be better to keep them.

> in the second case the interpolation surely makes sense,

Depending on the period and the speed, the accuracy can be very poor. But I guess that the user's choice. 

Agreed.